[Docs] Add explanations on debugging and re-loading to the run methods (#322)
mckinsey/vizro: vizro-core/docs/pages/user-guides/run.md

 * Debug mode: on
```

??? info "Automatic reloading and debugging"

>     #### Reloading the dashboard

Reviewer: ```suggestion
```

Reviewer: Having subsections in a callout adds them to the page navigation which feels a bit weird to me.
![image](https://github.com/mckinsey/vizro/assets/49395058/25b47033-425f-4942-a743-a072394dc5cd)

Probably we can turn this feature off somehow, but I actually think this callout doesn't need two different sections anyway so I'd suggest removing it.

    You can set up the front-end to automatically refresh whenever dashboard configuration updates are made, as described in the
    ["Code Reloading and Hot Reloading" section of the Dash Dev Tools documentation](https://dash.plotly.com/devtools#code-reloading-&-hot-reloading).

>     It is turned off by default in Vizro apps but can be enabled by using `debug=True` in the `run()` method, eg.

Reviewer: ```suggestion
    This is turned off by default in Vizro apps but can be enabled by using `debug=True` in the `run()` method, eg.
```

    `Vizro().build(dashboard).run(debug=True)`

>     (Note: this is the equivalent of enabling Dash Dev Tools via the `app.run()` method. See the [Dash documentation](https://dash.plotly.com/devtools) for more information).

Reviewer: ```suggestion
Setting `debug=True` enables [Dash Dev Tools](https://dash.plotly.com/devtools). In addition to hot reloading, this enables several other features that are useful during development, such as detailed in-app error reporting.
```

>     #### Debugging
> 
>     Dash dev tools debug mode is turned on as described above, by using `debug=True` in the `run()` method.
> 
>     `Vizro().build(dashboard).run(debug=True)`
> 

Reviewer: ```suggestion
```

I'd suggest simplifying this by just adding a bit more detail to the above. At the moment it seems too repetitive since this just repeats the above section.

- you can specify `jupyter_mode="external"` and a link will be displayed to direct you to the localhost where the dashboard is running.
- you can use tab mode by `jupyter_mode="tab"` to automatically open the app in a new browser

??? info "Reloading and debugging"

>     #### Reloading the dashboard
> 
>     Code reloading and hot reloading do not work from a Jupyter notebook, even if Dash Dev Tools debug mode is enabled.
>     You must restart the entire Jupyter kernel to reload the dashboard and reflect changes in the dashboard configuration.
> 
>     #### Debugging
> 
>     Dash dev tools debug mode is turned on as described above, by using `debug=True` in the `run()` method.
> 
>     `Vizro().build(dashboard).run(debug=True)`
> 
> 
>     (Note: this is the equivalent of enabling Dash Dev Tools via the `app.run()` method as described in the [Dash documentation](https://dash.plotly.com/devtools)).
> 
>     In addition, some errors generated at run time can also be viewed via the browser console (for example in `Chrome` see `View > Developer > Developer Tools > Console`).
> 
> 

Reviewer: ```suggestion
     When working in a Jupyter notebook, only some of the [Dash Dev Tools](https://dash.plotly.com/devtools) functionality is enabled by using `run(debug=True)`.
     In particular, code reloading and hot reloading do not work from a Jupyter notebook. Instead, you must restart the entire Jupyter kernel to reload the dashboard and reflect changes in the dashboard configuration.
```
